# Fractional Knapsack Problem (Greedy Algorithm)


Given weights and values of items, find the maximum value
that can be obtained by allowing fractional items.


In [7]:
def knapsackproblem():
    weights=[100,220,120,80,150]
    values=[10,20,30,40,50]
    curr=[]
    total_value=0
    capacity=int(input("Enter the amount of capacity:"))
    n=len(weights)
    for i in range(n):
        ratio=values[i]/weights[i]
        curr.append((ratio,values[i],weights[i]))
    curr.sort(reverse=True)
    for ratio,values,weights in curr:
        if weights<=capacity:
            capacity-=weights
            total_value+=values        
        else:
            total_value+=ratio*capacity
            break
    print("Maximum value:",total_value)
knapsackproblem()
        

Maximum value: 107.5


**Input:**
Capacity = 300

**Output:**
Maximum value = 107.5


The algorithm sorts items based on value/weight ratio and
selects items greedily to maximize total value.
